### PR TITLE
Handle content-type header charset value for streaming API

### LINKED ISSUE #615: TypeError: Can't convert 'bytes' object to str implicitly
I was following the streaming api tutorial and got this error. Other threads say it has been fixed.
Using master branch code.

typeerror

### PR DESCRIPTION
**Pull request by varzac**
I tested on python 3.4 and python 2.7.

I was unable to get the tox tests properly running on my machine.  This should fix #615 properly as well as properly use the content-type charset http header if it is present.

Closes #615

### CODE REVIEW

**Comment by swstack** [comment]
If you put "closes #615"  in your PR body it will close the issue automagically when this is merged.

File: tweepy/streaming.py [hunk 1]
```diff
@@ -168,7 +170,7 @@ def read_line(self, sep='\n'):
                 return self._pop(loc + len(sep))
             else:
                 start = len(self._buffer)
-            self._buffer += self._stream.read(self._chunk_size)
+            self._buffer += self._stream.read(self._chunk_size).decode(self._encoding)
```
Comment (posborne) [inline tweepy/streaming.py]: I'm not sure it is safe to perform decoding on arbitrary chunks of data as they come in as you could end up in a situation where you are using a valid decode on incomplete data.  Take for instance the following (python3.4):

```
>>> skull = "☠"
>>> skull_encoded = skull.encode('utf-8')
>>> skull_encoded
b'\xe2\x98\xa0'
>>> skull_encoded[:-1].decode('utf-8')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 0-1: unexpected end of data
```

Comment (varzac) [inline tweepy/streaming.py]: That is a good point and one that I didn't really consider.  How do you feel about doing the decode as a part of the pop function.  I.e. keep everything internal to the ReadBuffer as bytes, but decode on returns.  The other option of course is to just keep everything in bytes and return bytes, and just offload the decoding to the person using ReadBuffer.  What is your opinion?

Comment (posborne) [inline tweepy/streaming.py]: I think doing it on pop probably makes sense.

**Comment by danielquinn** [comment]
:+1:

**Pull request merged**